crates/forge/src/test_filter.rs

```diff
@@ -114,13 +140,13 @@ mod tests {
     #[test]
```

Comment:
Let's add a test where only_failed is `true`.

Reply:
I'm not sure, what do you mean 🤔

Reply:
You've added a new attribute to tests filter, but there's no unit test covering it.

Reply:
should we test it in the unit? We have already covered it in e2e tests. The problem with this test is that it's impossible to write it without ingeration in production code

Reply:
The amount of code required for mocking doesn't seem to be worth it. Let's make sure we have comprehensive e2e tests instead.